packages/wrangler/src/__tests__/publish.test.ts

```diff
@@ -1930,6 +1935,220 @@ export default{
+
+        test-name.test-sub-domain.workers.dev"
+      `);
+      expect(std.err).toMatchInlineSnapshot(`""`);
+      expect(std.warn).toMatchInlineSnapshot(`
+        "Deprecation notice: The \`build.upload.rules\` config field is no longer used, the rules should be specified via the \`rules\` config field. Delete the \`build.upload\` field from the configuration file, and add this:
+
+        [[rules]]
+        type = \\"Text\\"
+        globs = [ \\"**/*.file\\" ]
+        fallthrough = true
+        "
+      `);
+    });
+
+    it("should be able to use fallthrough:true for multiple rules", async () => {
+      writeWranglerToml({
+        rules: [
+          { type: "Text", globs: ["**/*.file"], fallthrough: true },
+          { type: "Text", globs: ["**/*.other"], fallthrough: true },
+        ],
+      });
+      fs.writeFileSync(
+        "./index.js",
+        `import TEXT from './text.file'; import OTHER from './other.other'; export default {};`
+      );
+      fs.writeFileSync("./text.file", "SOME TEXT CONTENT");
+      fs.writeFileSync("./other.other", "SOME OTHER TEXT CONTENT");
+      mockSubDomainRequest();
+      mockUploadWorkerRequest({
+        expectedType: "esm",
+        expectedBindings: [],
+        expectedModules: {
+          "./2d91d1c4dd6e57d4f5432187ab7c25f45a8973f0-text.file":
+            "SOME TEXT CONTENT",
+          "./16347a01366873ed80fe45115119de3c92ab8db0-other.other":
+            "SOME OTHER TEXT CONTENT",
+        },
+      });
+      await runWrangler("publish index.js");
+      expect(std.out).toMatchInlineSnapshot(`
+        "Uploaded
+        test-name
+        (TIMINGS)
+        Published
+        test-name
+        (TIMINGS)
+
+        test-name.test-sub-domain.workers.dev"
+      `);
+      expect(std.err).toMatchInlineSnapshot(`""`);
+      expect(std.warn).toMatchInlineSnapshot(`""`);
+    });
+
+    it("should be able to use fallthrough:false for multiple rules", async () => {
+      writeWranglerToml({
+        rules: [
+          { type: "Text", globs: ["**/*.file"], fallthrough: false },
+          { type: "Text", globs: ["**/*.other"] },
+        ],
+      });
+      fs.writeFileSync(
+        "./index.js",
+        `import TEXT from './text.file'; import OTHER from './other.other'; export default {};`
+      );
+      fs.writeFileSync("./text.file", "SOME TEXT CONTENT");
+      fs.writeFileSync("./other.other", "SOME OTHER TEXT CONTENT");
+
+      // We throw an error when we come across a file that matched a rule
+      // but was skipped because of faullthrough = false
+      let err: Error | undefined;
+      try {
+        await runWrangler("publish index.js");
+      } catch (e) {
+        err = e as Error;
+      }
+      expect(err?.message).toMatch(
+        `The file ./other.other matched a module rule in your configuration ({"type":"Text","globs":["**/*.other"]}), but was ignored because a previous rule with the same type had \`fallthrough = false\`.`
+      );
```

Comment:
because the full error contains a file path specific to the machine (so in my case, it contained my home dir etc) and I needed to match only with part of the error.

Reply:
Hmm. So I tried `rejects.toThrowError()` with the same string you provide here and it worked; so I am not sure what the problem is.

To be clear I tried:

```ts
      await expect(runWrangler("publish index.js")).rejects.toThrowError(
        `The file ./other.other matched a module rule in your configuration ({"type":"Text","globs":["**/*.other"]}), but was ignored because a previous rule with the same type had \`fallthrough = false\`.`
      );
```

Reply:
oh damn you're right, I was trying toMatchInlineSnapshot